core/src/dsnp/api_types.rs

```diff
@@ -50,17 +67,35 @@ pub trait Config {
 }
 
 /// Encapsulates all the keys and page data that needs to be retrieved from chain
-pub struct Import<E: EncryptionBehavior> {
+pub struct ImportBundle<E: EncryptionBehavior> {
 	/// graph owner dsnp user id
 	pub dsnp_user_id: DsnpUserId,
 
+	/// Connection type of graph being imported
+	pub connection_type: ConnectionType,
+
 	/// graph keys associated with this graph which is used for encryption and PRI generation
 	pub keys: Vec<KeyPair<E>>,
 
 	/// Raw page data containing the social graph retrieved from chain
 	pub pages: Vec<PageBlob>,
 }
 
+/// Encapsulates all page updates for a user's graph to be sent to the chain
+pub struct ExportBundle {
+	/// Graph owner DSNP user ID
+	pub dsnp_user_id: DsnpUserId,
+
+	/// Connection type of graph being exported
+	pub connection_type: ConnectionType,
+
+	/// Pages to be updated in the graph
+	pub updated_pages: Vec<PageBlob>,
+
+	/// Pages to be removed from the graph
+	pub removed_pages: Vec<PageId>,
```

Comment:
I think for removing a page you would need the previous hash. And we have another update type which is **AddKey**.  This works fine but we can use enums also to specify different types. Enum would be more Rusty but for C style apis we might need to change them to something like this. So overall I don't have any preference since each option have it's own pros and cons.

Reply:
Updating...